### PR TITLE
Fix some error checking for detailed schedules

### PR DESCRIPTION
**Pull request by shorowit**
## Pull Request Description

The documentation says that you must use MAX=1 for fractional schedules, but the code was only checking that the max was <= 1. Also fixes the battery schedule error message.

## Checklist

PR Author: Check these when they're done. Not all may apply. ~~strikethrough~~ and check any that do not apply. 

PR Reviewer: Verify each has been completed.

- [x] ~Schematron validator (`EPvalidator.xml`) has been updated~
- [x] ~Sample files have been added/updated (`openstudio tasks.rb update_hpxmls`)~
- [x] Tests have been added/updated (e.g., `HPXMLtoOpenStudio/tests/test*.rb` and/or `workflow/tests/test*.rb`)
- [x] ~Documentation has been updated~
- [x] Changelog has been updated
- [x] `openstudio tasks.rb update_measures` has been run
- [x] No unexpected changes to simulation results of sample files

### CODE REVIEW

File: .github/workflows/config.yml [hunk 1]
```diff
@@ -121,37 +121,46 @@ jobs:
...
+  merge-results:
+    runs-on: ubuntu-latest
+    needs: [run-workflow1-tests, run-workflow2-tests, run-unit-tests]
+    steps:
+      - name: Merge Artifacts
+        uses: actions/upload-artifact/merge@v4
+        with:
+          name: results
+          pattern: results-*
+          delete-merged: true
```
Comment (shorowit) [inline .github/workflows/config.yml:133]: The `upload-artifact` action no longer allows automatic merging of artifacts, instead they now require [using a separate action to achieve this](https://github.com/actions/upload-artifact/blob/main/docs/MIGRATION.md#merging-multiple-artifacts).

File: HPXMLtoOpenStudio/resources/schedules.rb [hunk 2]
```diff
@@ -618,8 +618,7 @@ def self.annual_equivalent_full_load_hrs(modelYear, schedule)
     # Get a 365-value array of which schedule is used on each day of the year,
     day_schs_used_each_day = schedule.getActiveRuleIndices(year_start_date, year_end_date)
     if !day_schs_used_each_day.length == 365
-      OpenStudio::logFree(OpenStudio::Error, 'openstudio.standards.ScheduleRuleset', "#{schedule.name} does not have 365 daily schedules accounted for, cannot accurately calculate annual EFLH.")
-      return 0
+      fail "#{schedule.name} does not have 365 daily schedules accounted for, cannot accurately calculate annual EFLH."
```
Comment (shorowit) [inline HPXMLtoOpenStudio/resources/schedules.rb:621]: Double-checking that we aren't hitting this error.

File: HPXMLtoOpenStudio/resources/schedules.rb [hunk 3]
```diff
@@ -672,11 +671,11 @@ def self.annual_equivalent_full_load_hrs(modelYear, schedule)
       annual_flh += daily_flh * number_of_days_sch_used
     end
 
-    # Warn if the max daily EFLH is more than 24,
+    # Check if the max daily EFLH is more than 24,
     # which would indicate that this isn't a
     # fractional schedule.
     if max_daily_flh > 24
-      OpenStudio::logFree(OpenStudio::Warn, 'openstudio.standards.ScheduleRuleset', "#{schedule.name} has more than 24 EFLH in one day schedule, indicating that it is not a fractional schedule.")
+      fail "#{schedule.name} has more than 24 EFLH in one day schedule, indicating that it is not a fractional schedule."
```
Comment (shorowit) [inline HPXMLtoOpenStudio/resources/schedules.rb:678]: Double-checking that we aren't hitting this either.

File: HPXMLtoOpenStudio/resources/schedules.rb [hunk 4]
```diff
@@ -1414,7 +1413,7 @@ def import(schedules_paths)
         end
 
         if max_value_one[col_name]
-          if values.max > 1
+          if values.max > 1.01 || values.max < 0.99 # Allow some imprecision
             fail "Schedule max value for column '#{col_name}' must be 1. [context: #{schedules_path}]"
```
Comment (shorowit) [inline HPXMLtoOpenStudio/resources/schedules.rb:1417]: Bugfix to match the documentation.

File: HPXMLtoOpenStudio/resources/schedules.rb [hunk 5]
```diff
@@ -1427,7 +1426,7 @@ def import(schedules_paths)
 
         if min_value_neg_one[col_name]
           if values.min < -1
-            fail "Schedule min value for column '#{col_name}' must be -1. [context: #{schedules_path}]"
+            fail "Schedule value for column '#{col_name}' must be greater than or equal to -1. [context: #{schedules_path}]"
```
Comment (shorowit) [inline HPXMLtoOpenStudio/resources/schedules.rb:1429]: Clarification.

File: docs/source/workflow_inputs.rst [hunk 6]
```diff
@@ -2211,10 +2211,7 @@ Each ``HeatPump`` is expected to represent a single outdoor unit, whether connec
          HeatingDetailedPerformanceData must also be provided.
   .. [#] If HeatingDetailedPerformanceData is provided, see :ref:`htg_detailed_perf_data`.
          CoolingDetailedPerformanceData must also be provided.
-  .. [#] If neither extension/HeatingCapacityRetention nor HeatingCapacity17F nor HeatingDetailedPerformanceData provided, heating capacity retention defaults based on CompressorType:
-         
-         \- **variable speed**: 0.0461 * HSPF + 0.1594 (at 5F)
-         
+  .. [#] If neither extension/HeatingCapacityRetention nor HeatingCapacity17F nor HeatingDetailedPerformanceData provided, heating capacity retention defaults to 0.0461 * HSPF + 0.1594 (at 5F).
```
Comment (shorowit) [inline docs/source/workflow_inputs.rst:2214]: Unrelated simplification.

File: HPXMLtoOpenStudio/resources/schedules.rb (hunk 4 above)
Comment (joseph-robertson) [reply HPXMLtoOpenStudio/resources/schedules.rb:1417]: Looks like `battery` is not excluded from this check. Should it be? If so, should it have a Units other than "frac" in the docs?

Comment (shorowit) [reply HPXMLtoOpenStudio/resources/schedules.rb:1417]: Good catch. Just addressed your comment.

**Review by joseph-robertson — APPROVED** [review]

**Pull request merged**